### PR TITLE
Feature/arc 1450 move query parameter

### PR DESCRIPTION
**Pull request by EmileVanTichelenSHD**
https://meemoo.atlassian.net/browse/ARC-1450

### CODE REVIEW

File: src/modules/ie-objects/controllers/ie-objects.controller.ts [hunk 1]
```diff
@@ -170,12 +172,13 @@ export class IeObjectsController {
 		return convertObjectToCsv(objectMetadata);
 	}
 
-	@Get(':schemaIdentifier/related/:meemooIdentifier')
+	@Get(':schemaIdentifier/related/:meemooIdentifier/')
```
Comment (tourlbr) [inline src/modules/ie-objects/controllers/ie-objects.controller.ts:175]: REQ: don't add slash at the end of this endpoint. I presume you are having an issue getting to this endpoint when others take over. This is not the solution. 

Q: Can you explain why ending slash was added here?

File: src/modules/ie-objects/controllers/ie-objects.controller.ts [hunk 2]
```diff
@@ -170,12 +172,13 @@ export class IeObjectsController {
...
 	public async getRelated(
 		@Headers('referer') referer: string,
 		@Param('schemaIdentifier') schemaIdentifier: string,
 		@Param('meemooIdentifier') meemooIdentifier: string,
-		@SessionUser() user: SessionUserEntity
+		@SessionUser() user: SessionUserEntity,
+		@Query() relatedQueryDto: RelatedQueryDto
```
Comment (tourlbr) [inline src/modules/ie-objects/controllers/ie-objects.controller.ts:181]: REQ:

1. Headers
2. Param
3. Query
4. other as in SessionUser

File: src/modules/ie-objects/controllers/ie-objects.controller.ts [hunk 3]
```diff
@@ -184,7 +187,8 @@ export class IeObjectsController {
 		const relatedIeObjects = await this.ieObjectsService.getRelated(
 			schemaIdentifier,
 			meemooIdentifier,
-			referer
+			referer,
+			relatedQueryDto.maintainerId
```
Comment (tourlbr) [inline src/modules/ie-objects/controllers/ie-objects.controller.ts:191]: REQ: just pass the dto as full object to the service and extract maintainerId inside service.

File: src/modules/ie-objects/controllers/ie-objects.controller.ts [hunk 4]
```diff
@@ -221,12 +225,17 @@ export class IeObjectsController {
 	public async getSimilar(
 		@Headers('referer') referer: string,
 		@Param('id') id: string,
-		@SessionUser() user: SessionUserEntity
+		@SessionUser() user: SessionUserEntity,
+		@Query() similarQueryDto: SimilarQueryDto
```
Comment (tourlbr) [inline src/modules/ie-objects/controllers/ie-objects.controller.ts:229]: REQ: same issue as mentioned above

File: src/modules/ie-objects/dto/ie-objects.dto.ts [hunk 5]
```diff
@@ -142,3 +142,23 @@ export class IeObjectMeemooIdentifiersQueryDto {
 	})
 	meemooIdentifiers: string[];
 }
+
+export class SimilarQueryDto {
```
Comment (tourlbr) [inline src/modules/ie-objects/dto/ie-objects.dto.ts:146]: REQ: Rename to IeObjectSimilarQueryDto

File: src/modules/ie-objects/dto/ie-objects.dto.ts [hunk 6]
```diff
@@ -142,3 +142,23 @@ export class IeObjectMeemooIdentifiersQueryDto {
...
+	@IsString()
+	@IsOptional()
+	@ApiPropertyOptional({
+		type: String,
+		description: 'Fetch similar object with the same maintainerId',
+	})
+	maintainerId?;
+}
+
+export class RelatedQueryDto {
```
Comment (tourlbr) [inline src/modules/ie-objects/dto/ie-objects.dto.ts:156]: REQ: Rename to IeObjectRelatedQueryDto

File: src/modules/ie-objects/services/ie-objects.service.ts [hunk 7]
```diff
@@ -138,14 +138,16 @@ export class IeObjectsService {
 	public async getRelated(
 		schemaIdentifier: string,
 		meemooIdentifier: string,
-		referer: string
+		referer: string,
+		maintainerId?: string
```
Comment (tourlbr) [inline src/modules/ie-objects/services/ie-objects.service.ts:142]: REQ: should be dto and then in function extract maintainerId to be used

**Review by tourlbr — CHANGES_REQUESTED** [review]

File: src/modules/ie-objects/controllers/ie-objects.controller.ts (hunk 1 above)
Comment (bertyhell) [inline src/modules/ie-objects/controllers/ie-objects.controller.ts:175]: Now that we figured out what the which endpoint is which, lets add some documentation to the swagger file:
```
	@ApiOperation({
		description:
			'Get objects that belong to the same meemoo identifier group. For instance objects that are highlights of the same soccer match.',
	})
```

Comment (EmileVanTichelenSHD) [reply src/modules/ie-objects/controllers/ie-objects.controller.ts:175]: I first tried adding the new parameter trough the url but found it it didn't work. I thought it didn't matter whether the slash is there or not. I am gonna change it.

File: src/modules/ie-objects/controllers/ie-objects.controller.ts [hunk 8]
```diff
@@ -221,12 +225,17 @@ export class IeObjectsController {
 	public async getSimilar(
```
Comment (bertyhell) [inline src/modules/ie-objects/controllers/ie-objects.controller.ts:225]: Now that we figured out which endpoint is which, lets add some documentation to the swagger file:
```
	@ApiOperation({
		description:
			'Get objects that cover the same subject as the passed object schema identifier. (elasticsearch: more like this)',
	})
```

File: src/modules/ie-objects/controllers/ie-objects.controller.ts (hunk 1 above)
Comment (tourlbr) [reply src/modules/ie-objects/controllers/ie-objects.controller.ts:175]: Adding it to url would mean it is a **Param** but we need a **Query** here. So yes please remove.

File: src/modules/ie-objects/controllers/ie-objects.controller.ts [hunk 9]
```diff
@@ -221,12 +225,17 @@ export class IeObjectsController {
...
+		@SessionUser() user: SessionUserEntity,
+		@Query() similarQueryDto: SimilarQueryDto
 	): Promise<IPagination<Partial<IeObject>>> {
 		const visitorSpaceAccessInfo =
 			await this.ieObjectsService.getVisitorSpaceAccessInfoFromUser(user);
 
-		const similarIeObjects = await this.ieObjectsService.getSimilar(id, '_all', referer);
+		const similarIeObjects = await this.ieObjectsService.getSimilar(
+			id,
+			similarQueryDto.maintainerId ? similarQueryDto.maintainerId.toLowerCase() : '_all',
```
Comment (bertyhell) [inline src/modules/ie-objects/controllers/ie-objects.controller.ts:236]: this should be the same behavior:
```suggestion
			similarQueryDto.maintainerId?.toLowerCase() ?? '_all',
```

**Review by bertyhell — CHANGES_REQUESTED** [review]

**Review by tourlbr — APPROVED** [review]

**Review by tourlbr — APPROVED** [review]

**Pull request merged**